src/main/java/com/adaptavist/tm4j/jenkins/cucumber/CucumberFileUtil.java

```diff
@@ -0,0 +1,86 @@
+package com.adaptavist.tm4j.jenkins.cucumber;
+
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystems;
+
+import static com.adaptavist.tm4j.jenkins.utils.Constants.INFO;
+
+public class CucumberFileUtil {
+
+    public static File filterCucumberFiles(File file, String directory, final PrintStream logger) {
```

Comment:
Is there any reason why we have to keep passing `logger` as a method argument instead of having it as a class field?

Reply:
keep passing logger because the methods are static.

Reply:
I did a refactor, to  not have logger as a method argument in all methods